node/node.go

```diff
@@ -234,9 +234,8 @@ func (node *Node) addPendingTransactions(newTxs types.Transactions) []error {
 
 // Add new staking transactions to the pending staking transaction list.
 func (node *Node) addPendingStakingTransactions(newStakingTxs staking.StakingTransactions) []error {
-	if node.NodeConfig.ShardID == shard.BeaconChainShardID &&
-		node.Blockchain().Config().IsPreStaking(node.Blockchain().CurrentHeader().Epoch()) {
-		poolTxs := types.PoolTransactions{}
+	if node.NodeConfig.ShardID == shard.BeaconChainShardID && node.isPreStakingEpoch() {
+		poolTxs := make(types.PoolTransactions, 0, len(newStakingTxs))
```

Comment:
Do not preallocate . This is wrong premature optimization

Reply:
Done. But could you tell me a bit about why it's wrong?

Reply:
You are trying to be smarter than compiler when it’s not necessary and preallocate lots of space bc this is txn pool which could potentially be big. Append is heavily optimized and pretty damn good on own.

Reply:
I am sorry I am little bit confused. According to my knowledge, append is done through double extending. In each double extending, a new sized array is allocated. So suppose txs size is `n` and it is big. If we allocate at the beginning, we allocate `n` pointers total. But if we do not use append, we allocate `2n` pointers in total, which is a waste in performace. Am I understanding this correctly?

Reply:
Member
No you aren’t. You don’t need to make these kinds of assumptions in a garbage collected language and it’s not necessary to think this deeply over this. Focus on business logic please

Reply:
Is there any related reading materials for recommendation for this topic?